src/utils/withWidth.d.ts

```diff
@@ -22,6 +23,6 @@ export function isWidthUp(
 
 export default function withWidth(
   options?: WithWidthOptions,
-): <P>(
-  component: React.ComponentType<P & WithWidthProps>,
-) => React.ComponentClass<P & Partial<WithWidthProps>>;
+): <P extends WithWidthProps>(
+  component: React.ComponentType<P>,
+) => React.ComponentClass<Omit<P, keyof WithWidthProps> & Partial<WithWidthProps>>;
```

Comment:
Similarly, this shouldn't require `P` to contain `width`:

```ts
export default function withWidth(
  options?: WithWidthOptions,
): <P extends Partial<WithWidthProps> & Record<string, any>>(
  component: React.ComponentType<P & WithWidthProps>,
) => React.ComponentClass<Omit<P, keyof WithWidthProps> & Partial<WithWidthProps>>;
```

With all of these HOCs we want to enable this kind of pattern:

```ts
const Component = withWidth()<{
  // shouldn't need to specify width here; it's a given
  name: string
}>(({ width, name }) =>
  <div style={{ width }}>hello, {name}</div>
);
```

Reply:
Yeah, I wondered about those two, should have hit them.  I'll add a couple of tests, too.